### ISSUE TITLE
UnboundLocalError exception thrown during cz bump

### ISSUE THREAD

**Issue by niall-byrne** [commitizen-tools/commitizen#1004]
### Description

When I run cz bump without any existing tags, an error is thrown if I have a pre_bump script configured.

This started as of version 3.16.0, versions before that had no issue.

### Steps to reproduce

1. Repo with no tags.
2. Configure a simple pre_bump script.
    ```bash
     #!/bin/bash
     # Standardize toml quotes
     set -eo pipefail
     main() {
        # sed compatible with Linux and BSD
        sed -i.bak "s,\"${CZ_PRE_NEW_VERSION}\",'${CZ_PRE_NEW_VERSION}',g" pyproject.toml
        rm pyproject.toml.bak
     }

     main
    ```

    ```toml
       [tool.commitizen]
       bump_message = 'bump(RELEASE): $current_version → $new_version'
       pre_bump_hooks = ['.cicd-tools/boxes/bootstrap/commitizen/pre_bump.sh']
    ```
4. cz bump --increment MINOR --yes

### Current behavior

```
poetry run cz bump --increment MINOR --yes
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    BRANCH_OR_TAG: master
    pythonLocation: /opt/hostedtoolcache/Python/[3](https://github.com/cicd-tools-org/cicd-tools/actions/runs/8175897168/job/22354095523#step:9:3).9.18/x6[4](https://github.com/cicd-tools-org/cicd-tools/actions/runs/8175897168/job/22354095523#step:9:4)
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.9.18/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.18/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.18/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.18/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.9.18/x64/lib
bump(RELEASE): 0.1.0 → 0.2.0
tag to create: 0.2.0
increment detected: MINOR

Traceback (most recent call last):
  File "/home/runner/.cache/pypoetry/virtualenvs/cicd-tools-3sbBasnr-py3.9/bin/cz", line 8, in <module>
    sys.exit(main())
  File "/home/runner/.cache/pypoetry/virtualenvs/cicd-tools-3sbBasnr-py3.9/lib/python3.9/site-packages/commitizen/cli.py", line [5](https://github.com/cicd-tools-org/cicd-tools/actions/runs/8175897168/job/22354095523#step:9:5)88, in main
    args.func(conf, arguments)()
  File "/home/runner/.cache/pypoetry/virtualenvs/cicd-tools-3sbBasnr-py3.9/lib/python3.9/site-packages/commitizen/commands/bump.py", line 34[6](https://github.com/cicd-tools-org/cicd-tools/actions/runs/8175897168/job/22354095523#step:9:6), in __call__
    is_initial=is_initial,
UnboundLocalError: local variable 'is_initial' referenced before assignment
Error: Process completed with exit code 1.
```

### Desired behavior

The version to be bumped a minor increment, and the pre_bump script to finish as it did before.

### Environment

This is happening both in my Linux CI and on my local OSX machine.

**Comment by woile** [comment]
@noirbizarre do you think it might be related to the new feature?

**Comment by noirbizarre** [comment]
I say it has been introduced by c245b14, by the `--exact` feature.

I'll submit a fix in the evening, it seems easy to fix. 
Maybe we are missing some tests for pre- and post-bump, I'll take a look at it too.

**Comment by noirbizarre** [comment]
There it is, reproduced and fixed in #1007
